src/app/modules/entity/entity-form/components/form-input/form-input.component.ts

```diff
@@ -126,7 +141,25 @@ export class FormInputComponent implements Field {
     }
   }
 
-  onInput(): void {
+  onInput(event: Event): void {
+    const inputValue = (event as MessageEvent).data;
+
+    if (this.config.inputType === 'number') {
+      if (this.config.errors) {
+        this.config.errors = null;
+      }
+
+      if (!this.shouldShowResetInput() && !inputValue !== null) {
```

Comment:
shouldShowResetInput() shows if input is valid, so we can show reset button-icon